### PR TITLE
TRUNK-5232 Clean up changedBy and dateChanged fields in domain object super classes and interfaces

### PR DESCRIPTION
**Pull request by aloomis85419**
Added all requested changes for the issue at https://issues.openmrs.org/browse/TRUNK-5232. Read the developer notes of this issue for further details.

## Issue I worked on

see https://issues.openmrs.org/browse/TRUNK-

## Checklist: I completed these to help reviewers :)

- [x] My pull request only contains **ONE single commit**
(the number above, next to the 'Commits' tab is 1).

  No? -> [read here](https://wiki.openmrs.org/display/docs/Pull+Request+Tips) on how to squash multiple commits into one

- [x] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [x] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [x] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [x] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [x] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

File: api/src/main/java/org/openmrs/util/Security.java [hunk 1]
```diff
@@ -33,7 +33,7 @@
  * OpenMRS's security class deals with the hashing of passwords.
  */
 public class Security {
-
+
```
Comment (kaweesi) [inline api/src/main/java/org/openmrs/util/Security.java]: always try to submit formatting changes in their own commit

File: api/src/main/java/org/openmrs/BaseChangeableOpenmrsData.java [hunk 2]
```diff
@@ -9,45 +9,53 @@
...
+	@Column(name = "date_changed")
+	private Date dateChanged;
 	
 	/**
 	 * @see Auditable#getChangedBy()
 	 */
 	@Override
 	public User getChangedBy() {
-		return super.getChangedBy();
+		return this.changedBy;
```
Comment (kaweesi) [inline api/src/main/java/org/openmrs/BaseChangeableOpenmrsData.java]: you may exclude this here

File: api/src/main/java/org/openmrs/BaseChangeableOpenmrsData.java [hunk 3]
```diff
@@ -9,45 +9,53 @@
...
 	}
 	
 	/**
 	 * @see Auditable#getDateChanged()
 	 */
 	@Override
-	public Date getDateChanged() {
-		return super.getDateChanged();
-	}
+	public Date getDateChanged() {return this.dateChanged;}
```
Comment (kaweesi) [inline api/src/main/java/org/openmrs/BaseChangeableOpenmrsData.java]: no need for 'this' here as well and other similar places where it was used in getters

**Comment by dkayiwa** [comment]
@aloomis85419 are you still working on this?

**Comment by dkayiwa** [comment]
@aloomis85419 ping me to reopen this when you resume working on it. 😊
